### PR TITLE
[MM][Perf] Add PaddleOCR-VL encoder CUDA graph support

### PR DESCRIPTION
**Pull request by harsha20032020**
Adds encoder (ViT) CUDA graph capture/replay support to
`PaddleOCRVLForConditionalGeneration`, implementing the
`SupportsEncoderCudaGraph` protocol introduced by the shared framework in
#42224.

Closes / contributes to #38175 (umbrella: encoder CUDA graph support for VLMs).

**Duplicate-work check** (per AGENTS.md — no open PR found):
```bash
gh pr list --repo vllm-project/vllm --state open --search "PaddleOCR cudagraph in:title,body"
gh pr list --repo vllm-project/vllm --state open --search "paddleocr encoder"
```

### Background

vLLM's LLM decoder already benefits from CUDA graphs, but the vision encoder
runs eagerly — its many small kernel launches (patch embed, attention, MLP ×27
layers) dominate time-to-first-token (TTFT) for image requests. Capturing the
encoder into a CUDA graph removes per-launch CPU dispatch overhead.

This PR follows the same pattern as the Qwen2-VL (#41736), Kimi-VL (#41992),
and Step3-VL (#42224) implementations under #38175.

### What changed and why

PaddleOCR-VL's SigLIP vision tower had four patterns that prevented CUDA graph
capture. Each required moving data-dependent work into a precompute step outside
the graph:

| Location | Problem | Fix |
|---|---|---|
| `SiglipVisionEmbeddings.forward` | Per-image `F.interpolate(size=(h,w))` — dynamic output shape per image | Precompute `position_embeddings` in `prepare_encoder_metadata`; graph path adds them with a fixed `+` op |
| `SiglipEncoder.forward` | Per-image Python loop + `pids.max()` host sync to build 2D RoPE | Precompute `rotary_pos_emb` and `max_seqlen` outside the graph; pass as optional args |
| `encode_image` | Builds `cu_seqlens` from Python lists, one image at a time | Replaced by a single batched path with precomputed, padded `cu_seqlens` |
| `Projector (mlp_AR)` | Per-image `rearrange("(t h p1 w p2) d → ...")` — reshape layout depends on `h,w` | Precompute a `merge_index` gather order; graph uses a static `index_select` + reshape (`Projector.forward_packed`) |

The projector runs **inside** the captured graph (covering the full SigLIP
encoder + spatial-merge projection). The existing per-image eager path is fully
preserved — all precomputed-buffer arguments are optional; `encoder_metadata=None`
runs the original logic unchanged with zero behaviour change when
`cudagraph_mm_encoder=False`.

The graph is automatically **disabled** for non-FA attention backends (e.g.
`TORCH_SDPA` on pre-SM80 GPUs) — `get_encoder_cudagraph_config` returns empty
`modalities` in that case and the manager falls back to eager transparently.
Image-only; no video support.

### Key new symbols

| Symbol | File | Role |
|---|---|---|
| `SiglipVisionModel.prepare_encoder_metadata` | `paddleocr_vl.py` | Eagerly builds all fixed-shape buffers: `position_embeddings`, `rotary_pos_emb`, `cu_seqlens` (padded), `max_seqlen`, `merge_index` |
| `Projector.forward_packed` | `paddleocr_vl.py` | Fixed-shape spatial merge via `merge_index` gather — the projector path captured in the graph |
| `_pad_paddleocr_cu_seqlens_buffer` | `paddleocr_vl.py` | Replay padding logic that keeps `cu_seqlens[-1]` (total patch count) invariant between capture and replay |
| `SupportsEncoderCudaGraph` protocol methods (×10) | `paddleocr_vl.py` | Full protocol implementation on `PaddleOCRVLForConditionalGeneration` |

---

## Test Plan

```bash
# 1. Numerical parity — metadata-driven graph path == per-image eager path
#    Runs on any CUDA GPU (fp32, ~5 s), no model download needed.
.venv/bin/python -m pytest tests/model_executor/test_paddleocr_vl_cudagraph.py -v

# 2. MRoPE regression — existing eager path unchanged
.venv/bin/python -m pytest tests/model_executor/test_paddleocr_vl_mrope.py -v

# 3. End-to-end capture/replay — dummy weights, no download, requires SM >= 8.0
.venv/bin/python -m pytest \
  "tests/models/multimodal/generation/test_vit_cudagraph.py::test_vit_cudagraph_image[paddleocr_vl]" \
  -v --timeout=300

# 4. TTFT latency benchmark (real model, A10G)
# Graph OFF — baseline
.venv/bin/vllm bench latency \
  --model PaddlePaddle/PaddleOCR-VL \
  --dtype bfloat16 \
  --max-model-len 8192 \
  --input-len 16 \
  --output-len 1 \
  --batch-size 8 \
  --limit-mm-per-prompt '{"image": {"count": 1, "width": 896, "height": 896}}' \
  --no-enable-prefix-caching \
  --mm-encoder-attn-backend FLASH_ATTN \
  -cc.cudagraph_mm_encoder=false

# Graph ON
.venv/bin/vllm bench latency \
  --model PaddlePaddle/PaddleOCR-VL \
  --dtype bfloat16 \
  --max-model-len 8192 \
  --input-len 16 \
  --output-len 1 \
  --batch-size 8 \
  --limit-mm-per-prompt '{"image": {"count": 1, "width": 896, "height": 896}}' \
  --no-enable-prefix-caching \
  --mm-encoder-attn-backend FLASH_ATTN \
  -cc.cudagraph_mm_encoder=true \
  -cc.encoder_cudagraph_token_budgets="[4096]"
```

---

## Test Results

**Hardware:** NVIDIA A10G (SM 8.6, 24 GB VRAM),
bfloat16, FlashAttention 2, batch=8, 896×896 images, `output-len=1`
(TTFT-focused: minimises decode time so encoder dominates measured latency).

Graph capture confirmed in logs:
```
Using AttentionBackendEnum.FLASH_ATTN for MMEncoderAttention.
EncoderCudaGraphManager initialized with budgets=[1024, 4096], max_batch_size=128
Encoder CUDA graph capture complete. Captured 2 budget graphs.
```

### TTFT latency

| Metric | Graph OFF | Graph ON | Improvement |
|---|---|---|---|
| Avg latency | 11.93 ms | 11.23 ms | **↓ 5.9%** |
| P50 latency | 11.93 ms | 11.20 ms | **↓ 6.1%** |
| P99 latency | 12.46 ms | 11.66 ms | **↓ 6.5%** |
| Iterations/s | 83.65 | 88.92 | **↑ 6.3%** |

### CODE REVIEW

**Comment by harsha20032020** [comment]
Hi @shen-shanshan , can you have a look at this?

**Review by shen-shanshan — COMMENTED** [review]
Sorry for late reply, I'm busy with working these days. Btw, could you please rebase on the latest main and resolve the conflicts?
